megamek/src/megamek/common/Crew.java

```diff
@@ -1354,10 +1385,64 @@ public boolean doConsoleRoleSwap() {
         return false;
     }
 
+    //region extraData
+    public void initializeExtraDataAsHashMap(int crewNumber) {
```

Comment:
Will every crew member have extra data? If not this should probably not assign a new hashmap for each crew member and just lazily initialize them in the add routines below.

Reply:
This is the primary use case for MekHQ-side implementation, as every crew member will have a given name and gender as the minimum use case, which makes this an easier way than lazily initializing it.